src/pages/SettingsView/domain/useSettingsView.ts

```diff
@@ -61,7 +66,15 @@ export function useSettingsView() {
         }
       },
       goBack() {
-        history.back();
+        if (!referrerRef.current) {
+          history.push(`${PLUGIN_BASE_URL}${ROUTES.PROFILES_EXPLORER_VIEW}`);
+          return;
+        }
+
+        const backUrl = new URL(referrerRef.current);
+        backUrl.searchParams.set('maxNodes', String(currentSettings.maxNodes));
```

Comment:
Because the referrer URL contains the previous max nodes value, we need to update it before going back.